### PR TITLE
 switch button for the lists and ability to create lists for chosen s…

### PR DESCRIPTION
**Pull request by aymanbakiri**
…ongs and top songs

### CODE REVIEW

File: app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModel.kt [hunk 1]
```diff
@@ -11,7 +11,13 @@ import javax.inject.Inject
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.launch
+import ch.epfl.cs311.wanderwave.model.data.Track
 
+
+
+
+// Define a simple class for a song list
+data class SongList(val name: String, val tracks: MutableList<Track> = mutableListOf())
```
Comment (KlaKalma) [inline app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModel.kt:20]: Why do you need that ? Unfortunately I think it's gonna be hard to adapt the firebase connection if you use a specific dataclass for a list of tracks. Is it important ? This is a completely open quesiton if you think I should implement the songlist (which you should rename playlist btw), then I'll do it.

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [versions]
-agp = "8.3.0"
+agp = "8.2.2"
```
Comment (KlaKalma) [inline gradle/libs.versions.toml:2]: Is is a merge error or it's actually something you need ?

File: app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModel.kt (hunk 1 above)
Comment (KlaKalma) [reply app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/ProfileViewModel.kt:20]: Now that I think about it, the more general choice should be :

- The data class is better if we want "independant" playlists, that live outside of a beacon, that are like use playlists
- Keeping a normal List is better for backend storage and on device storage. and good if we keep playlists for a user or for a beacon.

File: app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileScreen.kt [hunk 3]
```diff
@@ -42,29 +54,109 @@ val INPUT_BOX_NAM_SIZE = 150.dp
...
-  } else {
-
-    Column(modifier = Modifier.fillMaxSize().padding(16.dp).testTag("profileScreen")) {
-      Box(modifier = Modifier.fillMaxWidth()) {
-        VisitCard(Modifier, currentProfile)
-        ProfileSwitch(Modifier.align(Alignment.TopEnd), viewModel)
-        ClickableIcon(Modifier.align(Alignment.BottomEnd), Icons.Filled.Create)
-      }
+fun ProfileScreen(viewModel: ProfileViewModel) {
+    val currentProfileState by viewModel.profile.collectAsState()
```
Comment (mbouaissi) [inline app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileScreen.kt:58]: Here, I think what you should do is putting your code from line 85 to the end inside the Box that is line 78. Also, I don't think you should comment the EditableVisitCard here

Comment (mbouaissi) [reply app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileScreen.kt:58]: And also maybe try to divide your code in multiple function, so it's easier to see what part is doing what

**Review by mbouaissi — CHANGES_REQUESTED** [review]
I think your code is good and easily readable, because it's well commented :+1:. I just think that once he missunderstanding inside ProfileScreen will be solve, we would be able to merge it

File: gradle/libs.versions.toml (hunk 2 above)
Comment (aymanbakiri) [reply gradle/libs.versions.toml:2]: I have this error : The project is using an incompatible version (AGP 8.3.0) of the Android Gradle plugin. Latest supported version is AGP 8.2.2
So I was forced to change the agp from 8.3.0 to 8.2.2.
But I think that the problem comes from my android studio version so you can leave agp =8.3.0

**Review by mbouaissi — APPROVED** [review]
lgtm

**Review by mbouaissi — APPROVED** [review]
lgtm 2

**Pull request merged**